### PR TITLE
Fix op number constraint for readmitted patients

### LINKED ISSUE #7751: Remove all OP number validation
**Is your feature request related to a problem? Please describe.**
When a patient is readmitted to the same facility and both the consultation (new and previous) are Out-patient consultations (decision after consultation), the OP number may be the same as the last consultation's OP number. However, currently it throws an error 

**Describe the solution you'd like**

- [ ] Remove all validation for IP and OP fields

**Additional context**
image

cc: @nihal467 @aparnacoronasafe 

issue have been modified based on @gigincg and @aparnacoronasafe latest comments

### PR DESCRIPTION
**Pull request by shivankacker**
## Proposed Changes

- Fixes https://github.com/coronasafe/care_fe/issues/7751

## Merge Checklist

- [x] Tests added/fixed
- [x] Update docs in `/docs`
- [x] Linting Complete
- [x] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/facility/models/patient_consultation.py [hunk 1]
```diff
@@ -311,7 +311,8 @@ class Meta:
             models.UniqueConstraint(
                 fields=["patient_no", "facility"],
                 name="unique_patient_no_within_facility",
-                condition=models.Q(patient_no__isnull=False),
+                condition=models.Q(patient_no__isnull=False)
+                & models.Q(new_discharge_reason__isnull=False),
```
Comment (sainak) [inline care/facility/models/patient_consultation.py]: this wont work for more than 2 readmissions in the same facility,

**Review by sainak — CHANGES_REQUESTED** [review]

**Comment by sainak** [comment]
@skks1212 add test for the cases mentioned in the issue

**Comment by nihal467** [comment]
@skks1212 the PR is not working as expected

**Test Case** Status : failed

- [x] Same IP number for re-admitted patient within the facility { It should throw error }
- [x] Same OP number for re-admitted patient within the facility
- [x] Use Earlier IP number for re-admitted patient within the facility and his latest OP number for domiciliary care
- [ ] OP number is unique for different patient within the facility
- [ ] OP number can be modified to match previous consultation OP number


**Bug** 

image

- [ ] unable to create OP patients with same OP number

image

- [ ] When we re-admit same patient to the same facility with same IP number , it is throw "something went wrong" error, reuse the existing error shown for unique IP number

image

- [ ] when we re-admit same patient to the same facility using his earlier IP number (Admission) as the latest OP number (Domiciliary Care), it is throw "something "

**Comment by shivankacker** [comment]
PR was incomplete, forgot to label as wip 😅. Fixing now

**Comment by nihal467** [comment]
@skks1212 the PR is still having bugs, I can create multiple consultation with same IP number for a single patient within a facility

**Comment by nihal467** [comment]
As per the issue , we are removing Data validation on the OP No. field.

**Comment by nihal467** [comment]
LGTM

**Review by nihal467 — APPROVED** [review]

**Review by nihal467 — COMMENTED** [review]
when a patient is discharged, and we reuse the same IP for next patient, it is being accept

Note: When a patient is live, and we try to reuse the IP for a new patient, this error is thrown properly

**Review by sainak — COMMENTED** [review]
test case required

**Comment by nihal467** [comment]
LGTM

**Review by rithviknishad — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]

**Review by sainak — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]

**Comment by vigneshhari** [comment]
I still think we need a unique validator across patients, as in two live patients cannot have the same IP/OP number

**Pull request merged**